crates/ruma-events/tests/it/sticker.rs

```diff
@@ -79,7 +79,35 @@ fn content_deserialization() {
 
     let content = from_json_value::<StickerEventContent>(json_data).unwrap();
     assert_eq!(content.body, "Upload: my_image.jpg");
-    assert_eq!(content.url, "mxc://notareal.hs/file");
+    assert_matches!(content.source, StickerMediaSource::Plain(sticker_url));
+    assert_eq!(sticker_url, "mxc://notareal.hs/file");
+
+    let encrypted_json_data = json!({
+        "body": "Upload: my_image.jpg",
+        "file": {
+            "url": "mxc://notareal.hs/file",
+            "key": {
+                "kty": "oct",
+                "key_ops": ["encrypt", "decrypt"],
+                "alg": "A256CTR",
+                "k": "TLlG_OpX807zzQuuwv4QZGJ21_u7weemFGYJFszMn9A",
+                "ext": true
+            },
+            "iv": "S22dq3NAX8wAAAAAAAAAAA",
+            "hashes": {
+                "sha256": "aWOHudBnDkJ9IwaR1Nd8XKoI7DOrqDTwt6xDPfVGN6Q"
+            },
+            "v": "v2",
+        },
+        "info": {},
+    });
+
+    let encrypted_content = from_json_value::<StickerEventContent>(encrypted_json_data).unwrap();
```

Comment:
This should actually fail if the feature is not enabled. You should probably have something like that:

```rust
#[cfg(not(feature = "compat-encrypted-sticker"))]
{
  from_json_value::<StickerEventContent>(encrypted_json_data).unwrap_err();
}

#[cfg(feature = "compat-encrypted-sticker")]
{
  let encrypted_content = from_json_value::<StickerEventContent>(encrypted_json_data).unwrap();

  assert_eq!(encrypted_content.body, "Upload: my_image.jpg");
  assert_matches!(encrypted_content.source, StickerMediaSource::Encrypted(encrypted_sticker_url));
  assert_eq!(encrypted_sticker_url.url, "mxc://notareal.hs/file");
}